src/components/MeasurementsToolList.vue

```diff
@@ -30,48 +41,140 @@ const tools = computed(() => {
         axis,
       };
     });
+};
+
+// Flatten all tool types and add actions
+const tools = computed(() => {
+  return props.tools.flatMap(
+    ({ store, icon, details = MeasurementToolDetails }) => {
+      const toolsWithAxis = getTools(store);
+      return toolsWithAxis.map((tool) => ({
+        ...tool,
+        icon,
+        details,
+        remove: () => store.removeTool(tool.id),
+        jumpTo: () => store.jumpToTool(tool.id),
+        toggleHidden: () => {
+          const toggled = !store.toolByID[tool.id].hidden;
+          store.updateTool(tool.id, { hidden: toggled });
+        },
+        updateTool: (patch: Partial<AnnotationTool<string>>) => {
+          store.updateTool(tool.id, patch);
+        },
+      }));
+    }
+  );
 });
 
-const remove = (id: ToolID) => {
-  props.toolStore.removeTool(id);
-};
+// --- selection and batch actions  --- //
 
-const jumpTo = (id: ToolID) => {
-  props.toolStore.jumpToTool(id);
-};
+const toolIds = computed(() => tools.value.map((tool) => tool.id));
 
-const toggleHidden = (id: ToolID) => {
-  const toggled = !props.toolStore.toolByID[id].hidden;
-  props.toolStore.updateTool(id, { hidden: toggled });
-};
+const { selected, selectedAll, selectedSome, toggleSelectAll } =
+  useMultiSelection(toolIds);
+
+const forEachSelectedTool = (
+  callback: (tool: (typeof tools.value)[number]) => void
+) =>
+  selected.value
+    .map((id) => tools.value.find((tool) => id === tool.id))
+    .filter(nonNullable)
+    .forEach(callback);
+
+function removeAll() {
+  forEachSelectedTool((tool) => tool.remove());
+  selected.value = [];
+}
+
+const globalHidden = ref(false);
```

Comment:
This should instead be computed based on the aggregated state of the selection.
- If all selected tools are already hidden, it should be "show".
- If at least one selected too is visible, it should be "hide".

Reply:
Fixed.